TicTacToe2/TicTacToeGame.cs

```diff
@@ -32,24 +36,142 @@ private void DrawBoard()
 
         private void GetUserInput()
         {
-            //string player = "X";
-            //string choiceText;
-            //bool validInput = false;
+            string choiceText;
+            bool validInput = false;
+            bool legalMove = false;
+
+            while (!validInput && !legalMove)
+            {
+                try
+                {
+                    Console.WriteLine($"\nIt is player {player}'s turn.");
+                    Console.Write("Pick a number from the board to place your mark: ");
+                    choiceText = Console.ReadLine();
+
+                    bool isParsable = Int32.TryParse(choiceText, out int choice);
+                    if (isParsable)
+                    {
+                        if (choice <= 0 || choice > 9)
+                        {
+                            validInput = false;
+                            Console.WriteLine("Please enter a number between 1-9.");
+                        }
+                        else
+                        {
+                            choice--;
+                            if (space[choice] == "X" || space[choice] == "O")
+                            {
+                                choice++;
```

Comment:
There are a lot of nested brackets here. When that happens, I recommend splitting logic into different functions or using a different logical structure. In this case, just using this one function should be fine.

This is how I would describe how I expect this function to work:

- Until the user enters valid input
  - Get input from the user
  - If the input is not an integer, notify the player that an integer is expected
  - Else if the number is not in the appropriate range (1 to 9), notify the player that a number between 1 and 9 is expected
  - Else if the space indicated by the number is already taken, notify the player of that
  - Else the input is valid and the space indicated is marked with the player's symbol

Note: a try block should not be necessary